api/app/Http/Controllers/ApiController.php

```diff
@@ -43,6 +47,20 @@ public function getAll()
         return $this->getResponder()->collection($this->getRepository()->all());
     }
 
+    public function getAllPaginated(PaginatedRequestDecoratorInterface $request)
+    {
+        $count = $this->getRepository()->count();
+        $limit = $request->getLimit(static::PAGINATOR_DEFAULT_LIMIT, static::PAGINATOR_MAX_LIMIT);
+        $offset = $request->isGetLast()?$count-$limit:$request->getOffset();
+        //eloquent do not allow 0 or negative limit and offset
+        if ($limit <= 0) {
+            $collection = new Collection();
+        } else {
+            $collection = $this->getRepository()->all(['*'], $offset, $limit);
+        }
+        return $this->responder->paginatedCollection($collection, $offset, $count);
+    }
```

Comment:
Also, as demonstration can you implement this method for the get all articles route? Remember that there is customisation implementation as when a get all articles method is called, the `content` key must not be returned (it could be huge). Bear in mind that there might be some issue with the `exerpt` column as if that column is null then the content field must be returned.